### PR TITLE
New package Microsoft.DSC

### PR DESCRIPTION
**Pull request by Gijsreyn**
Checklist for Pull Requests
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Is there a linked Issue? #199292 

Manifests
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.9 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.9.0)?

image

Note: `<path>` is the directory's name containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/199293)

### CODE REVIEW

**Review by Dragon1573 — APPROVED** [review]
Manually verified.

<details><summary>Console logs</summary>

```text
--> Installing WinGet
--> Disabling safety warning when running installers
Tip: you can type 'Update-EnvironmentVariables' to update your environment variables, such as after installing a new software.

--> Configuring Winget
Enabled admin setting 'LocalManifestFiles'.
Enabled admin setting 'LocalArchiveMalwareScanOverride'.

--> Installing the Manifest 3.0.0-preview.12

Found Desired State Configuration [Microsoft.Dsc] Version 3.0.0-preview.12
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://github.com/PowerShell/DSC/releases/download/v3.0.0-preview.12/DSC-3.0.0-preview.12-x86_64-pc-windows-msvc.zip
  ██████████████████████████████  6.94 MB / 6.94 MB
Successfully verified installer hash
Extracting archive...
Successfully extracted archive
Starting package install...
Command line alias added: "dsc"
Path environment variable modified; restart your shell to use the new value.
Successfully installed

--> Refreshing environment variables

--> Comparing ARP Entries

DisplayName                 DisplayVersion   Publisher             ProductCode                  Scope
-----------                 --------------   ---------             -----------                  -----
Desired State Configuration 3.0.0-preview.12 Microsoft Corporation Microsoft.Dsc__DefaultSource User


PS C:\Users\WDAGUtilityAccount\Desktop\winget-pkgs> dsc.exe
Apply configuration or invoke specific DSC resources

Usage: dsc.exe [OPTIONS] <COMMAND>

Commands:
  completer  Generate a shell completion script
  config     Apply a configuration document
  resource   Invoke a specific DSC resource
  schema     Get the JSON schema for a DSC type
  help       Print this message or the help of the given subcommand(s)

Options:
  -l, --trace-level <TRACE_LEVEL>    Trace level to use [possible values: error, warn, info, debug, trace]
  -t, --trace-format <TRACE_FORMAT>  Trace format to use [possible values: default, plaintext, json]
  -h, --help                         Print help
  -V, --version                      Print version
```

</details>

**Comment by Gijsreyn** [comment]
@Dragon1573 Thanks for the approve. I have two questions you might be able to answer:

* Should we still rename the `PackageName` to have preview in there?
* Why is the build still failing if it is a preview? Shouldn't it be added in the preview folder?

**Comment by Dragon1573** [comment]
Hi @Gijsreyn 👋🏼,

> [!TIP]
>
> Link this pull request to the issue using ["close keywords"](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue) can prevent other contributors from creating duplicate pull requests. It can also guide bots to label the issue <https://github.com/microsoft/winget-pkgs/labels/In-PR>.

> [!IMPORTANT]
>
> "Close keywords" is only valid in **commit messages** or in [**PR details**](https://github.com/microsoft/winget-pkgs/pull/199293#issue-2744806443).

**Comment by Dragon1573** [comment]
> * Should we still rename the `PackageName` to have preview in there?

Yes, it is.

> * Why is the build still failing if it is a preview? Shouldn't it be added in the preview folder?

It seems I approve the PR by mistake. 🤣 I can't revoke it ...

Since I AM NOT one of the moderators, the PR will not be merged automatically.

File: manifests/m/Microsoft/Dsc/preview/3.0.0-preview.12/Microsoft.Dsc.locale.en-US.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Created using wingetcreate 1.9.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.9.0.schema.json
+
+PackageIdentifier: Microsoft.Dsc
```
Comment (Dragon1573) [inline manifests/m/Microsoft/Dsc/preview/3.0.0-preview.12/Microsoft.Dsc.locale.en-US.yaml:4]:
```suggestion
PackageIdentifier: Microsoft.Dsc.preview
```

File: manifests/m/Microsoft/Dsc/preview/3.0.0-preview.12/Microsoft.Dsc.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Created using wingetcreate 1.9.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.9.0.schema.json
+
+PackageIdentifier: Microsoft.Dsc
```
Comment (Dragon1573) [inline manifests/m/Microsoft/Dsc/preview/3.0.0-preview.12/Microsoft.Dsc.yaml:4]:
```suggestion
PackageIdentifier: Microsoft.Dsc.preview
```

**Comment by Gijsreyn** [comment]
Just added the preview text to the `PackageName`. I didn't totally get you on the preview folder. Should I move it back? :)

**Review by Dragon1573 — CHANGES_REQUESTED** [review]
Filenames of them should be migrated to `Microsoft.Dsc.preview.*`.

**Review by mdanish-kh — CHANGES_REQUESTED** [review]
Please remove `preview/3.0.0-preview.12` folder. Seems to be a duplicate

**Comment by Gijsreyn** [comment]
> Please remove `preview/3.0.0-preview.12` folder. Seems to be a duplicate

Oops, I was messing with the git global config settings because of capitalization. Should be good now!

**Review by mdanish-kh — CHANGES_REQUESTED** [review]
This will fail pipelines again since the PackageIdentifier does not match the yaml file names.

PackageId -> `Microsoft.Dsc.Preview` so the `.yaml` files should include the PackageId as `Microsoft.Dsc.Preview.installer.yaml` `Microsoft.Dsc.Preview.locale.en-US.yaml` and so on

Looping in @denelon for an appropriate PackageIdentifier for this package.

`Microsoft.DSC`, `Microsoft.PowerShell.DSC` `Microsoft.DSC.Preview`?

**Comment by mdanish-kh** [comment]
We should also be adding the `msix` installer as well

**Comment by Gijsreyn** [comment]
@mdanish-kh Are you talking about this one: https://apps.microsoft.com/detail/9pcx3hx4hz0z?hl=en-US&gl=GB ?

**Comment by mdanish-kh** [comment]
@Gijsreyn - it's the one included in the release artifacts. This one https://github.com/PowerShell/DSC/releases/download/v3.0.0-preview.12/DSC-3.0.0-preview.12-Win.msixbundle

**Comment by Gijsreyn** [comment]
Ah, gotcha, I already thought it was. Do you want me to attempt to include it?

**Comment by Gijsreyn** [comment]
Yes, was already on it. Should be added!

File: manifests/m/Microsoft/Dsc/Preview/3.0.0-preview.12/Microsoft.Dsc.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Created using wingetcreate 1.9.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.9.0.schema.json
+
+PackageIdentifier: Microsoft.Dsc.Preview
+PackageVersion: 3.0.0-preview.12
```
Comment (mdanish-kh) [inline manifests/m/Microsoft/Dsc/Preview/3.0.0-preview.12/Microsoft.Dsc.yaml:5]:
```suggestion
PackageVersion: 3.0.12.0
```

File: manifests/m/Microsoft/Dsc/Preview/3.0.0-preview.12/Microsoft.Dsc.installer.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# Created using wingetcreate 1.9.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.9.0.schema.json
+
+PackageIdentifier: Microsoft.Dsc.Preview
+PackageVersion: 3.0.12.0
```
Comment (stephengillie) [inline manifests/m/Microsoft/Dsc/Preview/3.0.0-preview.12/Microsoft.Dsc.installer.yaml:5]: Please update file path to match this line.

File: manifests/m/Microsoft/Dsc/Preview/3.0.0-preview.12/Microsoft.Dsc.locale.en-US.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Created using wingetcreate 1.9.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.9.0.schema.json
+
+PackageIdentifier: Microsoft.Dsc.Preview
+PackageVersion: 3.0.12.0
```
Comment (stephengillie) [inline manifests/m/Microsoft/Dsc/Preview/3.0.0-preview.12/Microsoft.Dsc.locale.en-US.yaml:5]: Please update file path to match this line.

File: manifests/m/Microsoft/Dsc/Preview/3.0.0-preview.12/Microsoft.Dsc.yaml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# Created using wingetcreate 1.9.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.9.0.schema.json
+
+PackageIdentifier: Microsoft.Dsc.Preview
+PackageVersion: 3.0.0-12.0
```
Comment (stephengillie) [inline manifests/m/Microsoft/Dsc/Preview/3.0.0-preview.12/Microsoft.Dsc.yaml:5]: Please update file path to match this line.

**Review by stephengillie — CHANGES_REQUESTED** [review]
Hi @Gijsreyn,

Automatic Validation ended with:
>Manifest: The manifest file path must match PackageIdentifier and PackageVersion properties from manifest. actual 
'manifests\m\Microsoft\Dsc\Preview\3.0.0-preview.12', expected 
'manifests\m\Microsoft\Dsc\Preview\3.0.12.0'

**Comment by Gijsreyn** [comment]
Hey @stephengillie! Should be good to go.

**Comment by denelon** [comment]
@Gijsreyn it looks like the package was added in: 
* https://github.com/microsoft/winget-pkgs/pull/199424

**Comment by stephengillie** [comment]
Close with reason: Duplicate of #199424;

**Comment by Gijsreyn** [comment]
Somebody snooping away my PRs @denelon. No worries, glad it's in. Thanks for the heads-up.

**Comment by denelon** [comment]
Lol, yeah... some people just love the WinGet :D

**Pull request closed without merging**